src/compiler/crystal/syntax/to_s.cr

```diff
@@ -198,17 +198,24 @@ module Crystal
     end
 
     def visit(node : Expressions)
+      parens = node.keyword == :"("
+
+      @str << "(" if parens
+
       if @inside_macro > 0
         node.expressions.each &.accept self
       else
-        node.expressions.each do |exp|
+        node.expressions.each_with_index do |exp, i|
           unless exp.nop?
             append_indent
             exp.accept self
-            newline
+            newline unless parens && i == node.expressions.size - 1
```

Comment:
It seems also hack 😆

Reply:
No... when `Expressions#keyword` is `:")"` then there's no need for a final newline.

There are other problems, `accept_with_maybe_begin` should be removed, but we also need `If` to know if it's a regular `if` or `a ? b : c`. 

At least this is a smaller diff and doesn't introduce a named arguments to fix it.